dvc/repo/brancher.py

```diff
@@ -77,7 +77,7 @@ def brancher(  # noqa: E302
         from dvc.fs import GitFileSystem
 
         for sha, names in found_revs.items():
-            self.__dict__.pop("index", None)
+            self._reset_cached_indecies()  # pylint: disable=W0212
```

Comment:
This is not actually required as the `self.fs` setter calls `Repo._reset()`.
See https://github.com/iterative/dvc/pull/8407#discussion_r992043019.

Anyway, the changes looks good to me.

Reply:
@skshetry I see, yes, in this case the main change is in the __init__ itself.  I was not sure if that is fine / expected that brancher resets SCM and state, tbh. So code can be simplified if that is the case. It seems it resets everything now twice.

I can do a follow up if you confirm that it's fine that we reset SCM.

There also must be a comment, it's very unexpected that `=` has a side effect like this.

Reply:
@skshetry Do we need a follow up on this?